lib/Mojolicious/Controller.pm

```diff
@@ -0,0 +1,886 @@
+
+  # Do something after the transaction has been finished
+  $c->on(finish => sub ($c) {
+    $c->app->log->debug('All data has been sent');
+  });
+
+  # Receive WebSocket message
+  $c->on(message => sub ($c, $msg) {
+    $c->app->log->debug("Message: $msg");
+  });
+
+  # Receive JSON object via WebSocket message
+  $c->on(json => sub ($c, $hash) {
+    $c->app->log->debug("Test: $hash->{test}");
+  });
+
+  # Receive WebSocket "Binary" message
+  $c->on(binary => sub ($c, $bytes) {
+    my $len = length $bytes;
+    $c->app->log->debug("Received $len bytes");
+  });
+
+=head2 param
+
+  my $value = $c->param('foo');
+  $c        = $c->param(foo => 'ba;r');
+  $c        = $c->param(foo => 'ba;r', 'baz');
+  $c        = $c->param(foo => ['ba;r', 'baz']);
+
+Access route placeholder values that are not reserved stash values, file uploads as well as C<GET> and C<POST>
+parameters extracted from the query string and C<application/x-www-form-urlencoded> or C<multipart/form-data> message
+body, in that order. If there are multiple values sharing the same name, and you want to access more than just the last
+one, you can use L</"every_param">. Parts of the request body need to be loaded into memory to parse C<POST>
+parameters, so you have to make sure it is not excessively large. There's a 16MiB limit for requests by default.
+
+  # Get first value
+  my $first = $c->every_param('foo')->[0];
+
+For more control you can also access request information directly.
+
+  # Only GET parameters
+  my $foo = $c->req->query_params->param('foo');
+
+  # Only POST parameters
+  my $foo = $c->req->body_params->param('foo');
+
+  # Only GET and POST parameters
+  my $foo = $c->req->param('foo');
+
+  # Only file uploads
+  my $foo = $c->req->upload('foo');
+
+=head2 render
+
+  my $bool = $c->render;
+  my $bool = $c->render(foo => 'bar', baz => 23);
+  my $bool = $c->render(template => 'foo/index');
+  my $bool = $c->render(template => 'index', format => 'html');
+  my $bool = $c->render(data => $bytes);
+  my $bool = $c->render(text => 'Hello!');
+  my $bool = $c->render(json => {foo => 'bar'});
+  my $bool = $c->render(handler => 'something');
+  my $bool = $c->render('foo/index');
+
+Render content with L<Mojolicious/"renderer"> and emit hooks L<Mojolicious/"before_render"> as well as
+L<Mojolicious/"after_render">, or dies if no response could be generated. All additional key/value pairs get merged into
+the L</"stash">.
+
+  # Render characters
+  $c->render(text => 'I ♥ Mojolicious!');
+
+  # Render characters (alternative)
+  $c->stash(text => 'I ♥ Mojolicious!')->render;
+
+  # Render binary data
+  use Mojo::JSON qw(encode_json);
+  $c->render(data => encode_json({test => 'I ♥ Mojolicious!'}));
+
+  # Render JSON
+  $c->render(json => {test => 'I ♥ Mojolicious!'});
+
+  # Render inline template
+  $c->render(inline => '<%= 1 + 1 %>');
+
+  # Render template "foo/bar.html.ep"
+  $c->render(template => 'foo/bar', format => 'html', handler => 'ep');
```

Comment:
`$r->get('/')->to('c#a');` ...
```
package M::C::C;
sub a ($self) {
   $self->render('b');
}
```
Since the controller is `C`, shouldn't a template of `b` be found in `templates/c`?  It seems not ideal for the action to need to know what folder to find its own (or sibling) template in.